src/main/kotlin/com/wynntils/athena/database/interfaces/DatabaseObject.kt

```diff
@@ -0,0 +1,41 @@
+package com.wynntils.athena.database.interfaces
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.wynntils.athena.database.DatabaseManager
+import org.litote.kmongo.Id
+import org.litote.kmongo.deleteOneById
+import org.litote.kmongo.getCollection
+
+@JsonIgnoreProperties(value = [ "table" ])
+interface DatabaseObject {
+
+    val table: String
+    val _id: Any
+
+    private fun save(): DatabaseObject{
+//        r.table(table).insert(this).optArg("conflict", "replace").runNoReply(DatabaseManager.connection)
+        DatabaseManager.db.getCollection<DatabaseObject>(table).insertOne(this)
+
+        return this
```

Comment:
Generally you shouldn't return `this` unless you're doing a builder pattern. The stdlib returns the same type of object for operations that create a new instance (think list.plus), and the result for operations that mutate (ex list.add).